### PR TITLE
Removing traces of nose

### PR DESCRIPTION
**Pull request by bpoldrack**
Addressing  #7100

(Need to figure how to deal with `py.path.local`s)

### CODE REVIEW

**Comment by yarikoptic** [comment]
FWIW -- I agree with the taken approach. Might have done a different check (that `(.parent / "tests" / "test_tests_utils.py").exists()`) but ok with suggsted. I would've only brought it into `maint` since @TheChymera would want it for a release, and it might come sooner from `maint`.

**Comment by bpoldrack** [comment]
> I would've only brought it into maint since @TheChymera would want it for a release

Agree, will do.

File: .github/workflows/test_extensions.yml [hunk 1]
```diff
@@ -60,12 +60,6 @@ jobs:
         case ${{ matrix.extension }} in
           datalad-metalad)
             DL_PIP_INSTALLS+=" nose";;
-          datalad-neuroimaging)
-            DL_TESTER=nose;;
-          datalad-hirni)
-            DL_REPO=psychoinformatics-de/${{ matrix.extension }}
-            DL_TESTER=nose
-            DL_NEED_SINGULARITY=1;;
```
Comment (yarikoptic) [inline .github/workflows/test_extensions.yml:68]: note: hirni is no longer tested and neuroimaging pytest-ified

**Comment by yarikoptic** [comment]
looks good to me, and we still have http://github.com/datalad/datalad/blob/HEAD/.github/workflows/test-nose.yml which tests that nose code with nose.  I will mark this as a candidate to cut the release (we need it) if merged.

> (Need to figure how to deal with `py.path.local`s)

is that still a "TODO" item? (there is no checkbox available/marked)

**Comment by bpoldrack** [comment]
> is that still a "TODO" item?

Nope. Ready from my point of view.

File: .github/workflows/test_extensions.yml (hunk 1 above)
Comment (bpoldrack) [reply .github/workflows/test_extensions.yml:68]: Yes, that's why I wanted to remove it, since this file kinda suggests a different reality.

**Comment by yarikoptic** [comment]
one (first) travis run got stuck... unclear why yet but unlikely, restarted for the sake of it. But given that the rest is all good -- let's proceed and cut release!

**Comment by yarikoptic** [comment]
doh - I was too late with adding `release` label :-/

**Comment by yarikoptic-gitmate** [comment]
PR released in [`0.9.4`](https://github.com/datalad/datalad/releases/tag/0.9.4)

**Comment by yarikoptic-gitmate** [comment]
PR released in [`0.17.9`](https://github.com/datalad/datalad/releases/tag/0.17.9)

**Pull request merged**